### PR TITLE
Fixes Issue 3296 -- LexicalPreservation bug for array brackets

### LINKED ISSUE #3296: LexicalPreservation bug for array brackets
Hello all, thank you for developing such a great tool. However, I found a bug when using `LexicalPreservation` tool when I try to add an annotation to a class field which has the type `array` and the brackets `[]` are behind the parameter's name as shown in the example below.

```java
class Test{
   String[] allTest;
}
```
Please note that the example below works fine and the exception happens only when the brackets are behind field name. 

```java
class Test{
   String allTest[];
}
```

I also provided the code below that will generate the same exception mentioned above when executed:
```java
String code = "public class Test { String[] allTest; }";
CompilationUnit cu = LexicalPreservingPrinter.setup(StaticJavaParser.parse(code));
Optional<ClassOrInterfaceDeclaration> clazzOptional = cu.getClassByName("Test");
assertTrue(clazzOptional.isPresent());
ClassOrInterfaceDeclaration clazz = clazzOptional.get();
clazz.getMembers().forEach(
        bodyDeclaration ->
                bodyDeclaration.ifFieldDeclaration(
                        fieldDeclaration -> {
                            NodeList<VariableDeclarator> vars =
                                    fieldDeclaration.asFieldDeclaration().getVariables();
                            for (VariableDeclarator v : vars) {
                                if (v.getName().toString().equals("allTest")) {
                                    fieldDeclaration.addMarkerAnnotation("Nullable");
                                    break;
                                }
                            }
                        }));
String changed = LexicalPreservingPrinter.print(cu);
```

Here is the error I get when executing the code above:

```java
java.lang.UnsupportedOperationException: kept child(VariableDeclarator) vs "["   <101>   (line 1,col 27)-(line 1,col 27)

	at com.github.javaparser.printer.lexicalpreservation.Difference.applyKeptDiffElement(Difference.java:513)
```

With best regards, 

Nima Karimipour

### PR DESCRIPTION
**Pull request by nimakarimipour**
Fixes #3296

### CODE REVIEW

**Comment by nimakarimipour** [comment]
@jlerbsc as requested, I amended the commits. I think this is ready for a review.

File: javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java [hunk 1]
```diff
@@ -510,7 +510,7 @@ private void applyKeptDiffElement(Kept kept, TextElement originalElement, boolea
                     originalIndex++;
                     diffIndex++;
                 } else {
-                    throw new UnsupportedOperationException("kept " + kept.getElement() + " vs " + originalElement);
+                    originalIndex++;
```
Comment (jlerbsc) [inline javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java:513]: Can you be more specific here before jp threw an exception but with your solution it always keep the token?

**Comment by MysterAitch** [comment]
Thank you @nimakarimipour for the report and PR! 

I agree that this fix works to solve the given test cases and doesn't _seem_ to break anything else... but it makes me nervous that I can't rationalise _why_ it works or when it might break :sweat_smile::sweat_smile:

This is a broader issue with `Difference.java` though, and I have added #3313 to tackle that more broadly.

**Comment by MysterAitch** [comment]
Ideally I would like to see branch conditions that specifically allow this case where the array brackets are in a different location. Perhaps it can be something similar to the handling of type arguments a little above it?

If this isn't forthcoming, given that tests pass, I lean towards merging this in as an incremental improvement with a `// TODO` comment linking to the issue and/or indicating that it needs to be verified.

**Comment by MysterAitch** [comment]
@jlerbsc / @nimakarimipour 

Given that #3316 has been merged, is this PR still needed? :)

**Comment by jlerbsc** [comment]
Hi @MysterAitch there is no need to merge this PR. Thank you @nimakarimipour for your proposal.

**Pull request closed without merging**